handler.go

```diff
@@ -24,6 +42,22 @@ func (s *server) handleToken(w http.ResponseWriter, r *http.Request) {
 		})
 	}
 
+	key := cookiesKey(cookies)
+
+	// --- Cache check ---
+	cacheMutex.Lock()
+	entry, exists := tokenCache[key]
+	if exists && time.Now().Before(entry.Expiry) {
+		slog.InfoContext(ctx, "Returning cached Spotify token for key", slog.String("key", key))
+		cacheMutex.Unlock()
+
+		w.Header().Set("Content-Type", "application/json")
+		_, _ = w.Write(entry.Token)
+		return
+	}
+	cacheMutex.Unlock()
```

Comment:
locking and unlocking here adds a bug where the token could be fetched multiple times if 2 requests happen basically at the same time